tools/bridge/end2end-tests/tests/test_transfer.py

```diff
@@ -485,3 +522,186 @@ def check_balance():
         assert home_balance_before - home_balance_after == -value
 
     assert_within_timeout(check_balance, 10)
+
+
+def test_offline_validators_validates_not_complete_transfer(
+    web3_home, web3_foreign, foreign_bridge_contract, token_contract, accounts, bridges
+):
+    """
+    Tests whether a transfer initiated when threshold validators are offline will be effective when enough validators turn online
+    """
+    bridges[0].start()
+
+    sender = accounts[3]
+    value = 5000
+    foreign_balance_before = token_contract.functions.balanceOf(sender).call()
+    home_balance_before = web3_home.eth.getBalance(sender)
+
+    wait_for_successful_transaction_receipt(
+        web3_foreign,
+        token_contract.functions.transfer(
+            foreign_bridge_contract.address, value
+        ).transact({"from": sender}),
+    )
+
+    mine_min_blocks(web3_foreign, REORG_DEPTH)
+
+    def check_balances_bridge_transfer_not_complete():
+        foreign_balance_transfer_not_complete = token_contract.functions.balanceOf(
+            sender
+        ).call()
+        home_balance_transfer_not_complete = web3_home.eth.getBalance(sender)
+
+        assert foreign_balance_before - foreign_balance_transfer_not_complete == value
+        assert home_balance_before == home_balance_transfer_not_complete
+
+    assert_within_timeout(check_balances_bridge_transfer_not_complete, 10)
+
+    bridges[1].start()
+
+    def check_balance_transfer_complete():
+        home_balance_transfer_complete = web3_home.eth.getBalance(sender)
+
+        assert home_balance_transfer_complete - home_balance_before == value
+
+    assert_within_timeout(check_balance_transfer_complete, 10)
+
+
+def test_offline_validators_do_not_validates_complete_transfer(
+    web3_home,
+    web3_foreign,
+    foreign_bridge_contract,
+    token_contract,
+    accounts,
+    bridges,
+    validators,
+):
+    """
+    Tests that a transfer initiated when threshold validators are online will not be pointlessly validated when different validator turn online
+    """
+    bridge_2_address = validators[2]
+    before_tx_count = web3_home.eth.getTransactionCount(bridge_2_address)
+
+    bridges[0].start()
+    bridges[1].start()
+
+    sender = accounts[3]
+    value = 5000
+    home_balance_before = web3_home.eth.getBalance(sender)
+
+    wait_for_successful_transaction_receipt(
+        web3_foreign,
+        token_contract.functions.transfer(
+            foreign_bridge_contract.address, value
+        ).transact({"from": sender}),
+    )
+
+    mine_min_blocks(web3_foreign, REORG_DEPTH)
+
+    def check_balance_transfer_complete():
+        home_balance_transfer_complete = web3_home.eth.getBalance(sender)
+        assert home_balance_transfer_complete - home_balance_before == value
+
+    # We need to wait for the transaction to be completed before starting bridges[2]
+    assert_within_timeout(check_balance_transfer_complete, 10)
+
+    bridges[2].start()
+
+    time.sleep(10)
```

Comment:
same as above. I would write a function `assert_after_timeout` so that you do not always have to write the sleep there.

Reply:
But then I have to write a whole function to give to the `assert_after_timeout`

Reply:
Yes. You can also leave it like this. I just think it would make it clearer what the sleep is doing there and communicate that we do not want sleeps for no reason